### PR TITLE
refactor internal modules

### PR DESCRIPTION
**Pull request by sokra**
### Description

* get rid of attached filesystem for our embedded modules
* get rid of import "." in favor of inner assets

depends on #3999

### Testing Instructions

existing tests

[X] Auto label

### CODE REVIEW

File: crates/next-core/src/next_client_chunks/with_chunks.rs [hunk 1]
```diff
@@ -91,6 +92,7 @@ struct WithChunksChunkItem {
     context: ChunkingContextVc,
     inner_context: ChunkingContextVc,
     inner: WithChunksAssetVc,
+    chunk_item: EcmascriptChunkItemVc,
```
Comment (jridgewell) [inline crates/next-core/src/next_client_chunks/with_chunks.rs:95]: It doesn't look like this is actually used?

File: crates/next-core/src/next_client_component/with_client_chunks.rs [hunk 2]
```diff
@@ -91,6 +93,7 @@ impl EcmascriptChunkPlaceable for WithClientChunksAsset {
 struct WithClientChunksChunkItem {
     context: ChunkingContextVc,
     inner: WithClientChunksAssetVc,
+    chunk_item: EcmascriptChunkItemVc,
```
Comment (jridgewell) [inline crates/next-core/src/next_client_component/with_client_chunks.rs:96]: Ditto not used?

**Review by jridgewell — APPROVED** [review]

File: crates/next-core/src/next_client/transition.rs [hunk 3]
```diff
@@ -83,18 +69,30 @@ impl Transition for NextClientTransition {
...
-        let chunkable_asset = match ChunkableAssetVc::resolve_from(asset).await? {
-            Some(chunkable_asset) => chunkable_asset,
-            None => bail!(
-                "asset {} is not chunkable",
-                asset.ident().to_string().await?
-            ),
+        let internal_asset = if self.is_app {
+            next_asset("entry/app/hydrate.tsx")
+        } else {
+            next_asset("entry/next-hydrate.tsx")
```
Comment (alexkirsz) [inline crates/next-core/src/next_client/transition.rs:75]: We should move `entry/next-hydrate.tsx` to `entry/page/hydrate.tsx` to make it symmetric with `entry/app`.

File: crates/next-core/src/next_client_component/server_to_client_transition.rs [hunk 4]
```diff
@@ -14,16 +25,44 @@ pub struct NextServerToClientTransition {
...
+            Value::new(ReferenceType::Entry(
+                EntryReferenceSubType::AppClientComponent,
+            )),
+        );
+        let client_module = context.with_transition("next-ssr-client-module").process(
+            asset,
+            Value::new(ReferenceType::Entry(
+                EntryReferenceSubType::AppClientComponent,
+            )),
+        );
```
Comment (alexkirsz) [inline crates/next-core/src/next_client_component/server_to_client_transition.rs:51]: It seems weird to put these transitions inside of the context, only to always apply them here. These `process` calls should just be methods on `NextServerToClientTransition`: e.g. `process_client_chunks`, `process_client_module`

File: crates/next-core/src/page_source.rs [hunk 5]
```diff
@@ -656,36 +651,60 @@ impl SsrEntryVc {
...
             ),
-            SsrType::Html => VirtualAssetVc::new(
-                this.entry_asset.ident().path().join("server-renderer.tsx"),
-                next_js_file("entry/server-renderer.tsx").into(),
+            SsrType::EdgeApi => {
+                let entry_asset_edge_chunk_group =
+                    this.context.with_transition("next-edge").process(
+                        this.entry_asset,
+                        Value::new(ReferenceType::Entry(EntryReferenceSubType::PagesApi)),
+                    );
```
Comment (alexkirsz) [inline crates/next-core/src/page_source.rs:667]: Same comment for applying this transition here: could we avoid adding it to the context altogether?

File: crates/next-core/src/page_source.rs [hunk 6]
```diff
@@ -656,36 +651,60 @@ impl SsrEntryVc {
...
+            }
+            SsrType::Data => (
+                next_asset("entry/server-data.tsx"),
+                indexmap! {
+                    "INNER".to_string() => entry_asset_page,
+                },
             ),
+            SsrType::Html => {
+                let entry_asset_client_chunk_group =
+                    this.context.with_transition("next-client").process(
```
Comment (alexkirsz) [inline crates/next-core/src/page_source.rs:683]: Same comment.

File: crates/turbopack-node/src/evaluate.rs [hunk 7]
```diff
@@ -78,6 +78,8 @@ pub async fn get_evaluate_pool(
     let file_name = module_path.file_name();
     let file_name = if file_name.ends_with(".js") {
         Cow::Borrowed(file_name)
+    } else if let Some(file_name) = file_name.strip_suffix(".ts") {
+        Cow::Owned(format!("{file_name}.js"))
```
Comment (alexkirsz) [inline crates/turbopack-node/src/evaluate.rs:82]: Any other extensions we should handle here? Like .jsx and .tsx?

File: crates/turbopack-node/src/pool.rs [hunk 8]
```diff
@@ -176,18 +176,21 @@ impl NodeJsPoolProcess {
                         match status {
                             Ok(status) => {
                                 let (stdout, stderr) = get_output(&mut child).await?;
+                                println!("node process exited before we could connect to it with {status}Process output:\n{stdout}\nProcess error output:\n{stderr}");
```
Comment (alexkirsz) [inline crates/turbopack-node/src/pool.rs:179]: I expect this was for debugging? Remove it or add it as an `eprintln`. Also, a `\n` disappeared between `{status}` and `Process output`

File: crates/turbopack-node/src/pool.rs [hunk 9]
```diff
@@ -176,18 +176,21 @@ impl NodeJsPoolProcess {
...
                                 bail!("node process exited before we could connect to it with {status}\nProcess output:\n{stdout}\nProcess error output:\n{stderr}");
                             }
                             Err(err) => {
                                 let _ = child.start_kill();
                                 let (stdout, stderr) = get_output(&mut child).await?;
+                                println!("node process exited before we could connect to it: {err:?}Process output:\n{stdout}\nProcess error output:\n{stderr}");
```
Comment (alexkirsz) [inline crates/turbopack-node/src/pool.rs:185]: Same comment.

File: crates/turbopack-node/src/pool.rs [hunk 10]
```diff
@@ -176,18 +176,21 @@ impl NodeJsPoolProcess {
...
                                 bail!("node process exited before we could connect to it: {err:?}\nProcess output:\n{stdout}\nProcess error output:\n{stderr}");
                             },
                         }
                     },
                     _ = sleep(timeout) => {
                         let _ = child.start_kill();
                         let (stdout, stderr) = get_output(&mut child).await?;
+                        println!("timed out waiting for the Node.js process to connect ({timeout:?})Process output:\n{stdout}\nProcess error output:\n{stderr}");
```
Comment (alexkirsz) [inline crates/turbopack-node/src/pool.rs:193]: Same comment.

File: crates/turbopack/src/resolve_options_context.rs [hunk 11]
```diff
@@ -17,7 +18,9 @@ pub struct ResolveOptionsContext {
     pub enable_typescript: bool,
     pub enable_react: bool,
     pub enable_node_native_modules: bool,
-    pub enable_node_modules: bool,
+    /// Enable resolving of the node_modules folder when withing the provided
+    /// directory
+    pub enable_node_modules: Option<FileSystemPathVc>,
```
Comment (alexkirsz) [inline crates/turbopack/src/resolve_options_context.rs:23]:
```suggestion
    pub node_modules_directory: Option<FileSystemPathVc>,
```
This should no longer read `enable_smth`, but be called `node_modules_directory` directly.

**Pull request merged**